app/models/aliquot.rb

```diff
@@ -24,7 +24,8 @@ class Aliquot < ApplicationRecord
 
   self.lazy_uuid_generation = true
 
-  TagClash = Class.new(ActiveRecord::RecordInvalid)
+  # TagClash = Class.new(ActiveRecord::RecordInvalid)
+  TagClash = Class.new(StandardError)
```

Comment:
What was the reason for changing the exception type it inherits from, out of interest?

Reply:
Ah, just seen James's comment on the issue, Rails was swallowing the RecordInvalid one?

Reply:
Yes, we think it was related with this: 
https://guides.rubyonrails.org/active_record_callbacks.html#halting-execution

Reply:
Ah OK, so saving a record through the API is like calling `save` rather than `save!` ?

Reply:
Yes, we think; so because we were raising something like a ActiveRecord::RecordInvalid inside a callback, callbacks do not seem to re-raise it because that would break save() execution, so it must be because api v1 behaves the same as save.